orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/ArtifactUtils.java

```diff
@@ -230,38 +201,10 @@ public List<Artifact> getArtifactsForPipelineIdWithoutStageRef(
         .orElse(Collections.emptyList());
   }
 
-  private List<String> getExpectedArtifactIdsFromMap(Map<String, Object> trigger) {
-    List<String> expectedArtifactIds = (List<String>) trigger.get("expectedArtifactIds");
-    return (expectedArtifactIds != null) ? expectedArtifactIds : emptyList();
-  }
-
   public void resolveArtifacts(Map pipeline) {
     Map<String, Object> trigger = (Map<String, Object>) pipeline.get("trigger");
-    List<?> triggers = Optional.ofNullable((List<?>) pipeline.get("triggers")).orElse(emptyList());
-    Set<String> expectedArtifactIdsListConcat =
-        new HashSet<>(getExpectedArtifactIdsFromMap(trigger));
-
-    // Due to 8df68b79cf1 getBoundArtifactForStage now does resolution which
-    // can potentially return null artifact back, which will throw an exception
-    // for stages that expect a non-null value. Before this commit, when
-    // getBoundArtifactForStage was called, the method would just retrieve the
-    // bound artifact from the stage context, and return the appropriate
-    // artifact back. This change prevents tasks like CreateBakeManifestTask
-    // from working properly, if null is returned.
-    //
-    // reference: https://github.com/spinnaker/orca/pull/4397
-    triggers.stream()
```

Comment:
So this PR addresses an earlier PR, but we can also revert it, because the earlier PR that was submitted only implemented a partial solution. I've talked some about this offline to @nemesisOsorio, but we can revert this now since it does break multiple triggers of the same time. We will need to implement a proper fix

Reply:
Is the plan to create the proper fix in another PR?

Reply:
Yea, Ill have a PR that should address the issues with artifact resolution. I think Im going to write a proposal first, since some of the ideas I have require some refactors. So want to make sure we get that write in design first :)